flexmeasures/data/migrations/versions/cb8df44ebda5_flexcontext_field.py

```diff
@@ -295,18 +320,19 @@ def downgrade():
 
     for row in results:
         asset_id, flex_context, attributes_data = row
-        consumption_price_sensor_id = (
-            flex_context.get("consumption-price")["sensor"]
-            if flex_context.get("consumption-price")
-            else None
+
+        consumption_price_as_str, consumption_price_sensor_id = get_price_info(
+            flex_context.get("consumption-price")
         )
-        production_price_sensor_id = (
-            flex_context.get("production-price")["sensor"]
-            if flex_context.get("production-price")
-            else None
+        production_price_as_str, production_price_sensor_id = get_price_info(
+            flex_context.get("production-price")
         )
 
-        market_id = consumption_price_sensor_id
+        if consumption_price_sensor_id is not None:
+            market_id = consumption_price_sensor_id
+        else:
+            market_id = None
+
         capacity_in_mw = (
```

Comment:
This variable name is a source of confusion, I believe. At this point in the downgrade, it can contain a fixed capacity (quantity written as a string) or a sensor reference. If it's a sensor reference, we won't be able to convert it to a fixed capacity in MW. So in that case, we shouldn't store it under the `capacity_in_mw` attribute. I suggest two actions:

1. Rename the variable (in the downgrade command), for clarity. For example, `site_power_capacity`.
2. In case of anything other than a string power quantity, store the value under attributes using its original field name: `site-power-capacity`.